eslint.config.mjs

```diff
@@ -1,61 +1,37 @@
-import { fixupConfigRules, fixupPluginRules } from '@eslint/compat'
 import react from 'eslint-plugin-react'
 import reactHooks from 'eslint-plugin-react-hooks'
 import testingLibrary from 'eslint-plugin-testing-library'
 import jest from 'eslint-plugin-jest'
 import jestDom from 'eslint-plugin-jest-dom'
-import typescriptEslint from '@typescript-eslint/eslint-plugin'
-import tsParser from '@typescript-eslint/parser'
-import path from 'node:path'
-import { fileURLToPath } from 'node:url'
-import js from '@eslint/js'
-import { FlatCompat } from '@eslint/eslintrc'
+import barrel from 'eslint-plugin-barrel-files'
+import eslint from '@eslint/js'
+import tseslint from 'typescript-eslint'
+import storybook from 'eslint-plugin-storybook'
 
-const __filename = fileURLToPath(import.meta.url)
-const __dirname = path.dirname(__filename)
-const compat = new FlatCompat({
-  baseDirectory: __dirname,
-  recommendedConfig: js.configs.recommended,
-  allConfig: js.configs.all,
-})
-
-export default [
-  ...fixupConfigRules(
-    compat.extends(
-      'eslint:recommended',
-      'plugin:react/recommended',
-      'plugin:jest/recommended',
-      'plugin:@typescript-eslint/recommended-type-checked',
-      'plugin:storybook/recommended',
-    ),
-  ),
+export default tseslint.config(
+  eslint.configs.recommended,
+  ...tseslint.configs.recommendedTypeChecked,
+  ...storybook.configs['flat/recommended'],
   {
     plugins: {
-      react: fixupPluginRules(react),
-      'react-hooks': fixupPluginRules(reactHooks),
-      'testing-library': testingLibrary,
-      jest: fixupPluginRules(jest),
-      'jest-dom': jestDom,
-      '@typescript-eslint': fixupPluginRules(typescriptEslint),
+      react,
+      'react-hooks': reactHooks,
+      jest,
+      jestDom,
+      testingLibrary,
+      barrel,
     },
-    languageOptions: {
-      globals: {
-        JSX: true,
+    settings: {
+      react: {
+        version: '18',
       },
-      parser: tsParser,
-      ecmaVersion: 5,
-      sourceType: 'script',
+    },
+    languageOptions: {
       parserOptions: {
-        project: [
-          './tsconfig.json',
-          './packages/**/tsconfig.json',
-          './apps/**/tsconfig.json',
-        ],
+        projectService: true,
```

Comment:
This `projectService: true` is new functionality in typescript-eslint, and is supposedly much faster.